src/lib.rs

```diff
@@ -326,10 +333,14 @@ pub struct Evaluator<'tcx> {
 
     /// Stacked Borrows state.
     pub(crate) stacked_borrows: stacked_borrows::State,
+
+    /// The random number generator to use if Miri
+    /// is running in non-deterministic mode
+    pub(crate) rng: Option<RefCell<StdRng>>
```

Comment:
Why is this in a `RefCell`?

Reply:
`StdRng.fill_bytes` takes `&mut self`, but we're immutably borrowing `Evaluator` when we need to call it.

Reply:
`emulate_foreign_item` takes `&mut self`. Why should we have only an immutable borrow?